usr/sbin/p11sak/p11sak.c

```diff
@@ -1141,6 +1191,200 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
     }
 }
 
+/**
+ *  Print in p11sak_defined_attrs.conf defined attributes in short format
+ */
+static CK_RV short_print_vendor() 
+{
+    CK_RV rc = CKR_OK;
+    int f;
+    FILE *fp;
+    struct ConfigBaseNode *cfg, *c, *name, *hex_string, *type;
+    struct ConfigStructNode *structnode;
+    char file_loc[1024] = P11SAK_DEFINED_ATTRS_LOCATION;
+
+    // get file location
+    cfg = NULL;
+    if (strcmp(P11SAK_DEFINED_ATTRS_LOCATION, "") == 1) {
+        strcpy(file_loc, P11SAK_DEFINED_ATTRS_LOCATION);
+    } else {
```

Comment:
I don't get the sense of P11SAK_DEFINED_ATTRS_LOCATION....

Also you initialize file_loc with P11SAK_DEFINED_ATTRS_LOCATION above already, so the strcpy is superfluous.

I would actually always use P11SAK_DEFAULT_CONF_FILE and remove P11SAK_DEFINED_ATTRS_LOCATION  completely.
If you want to make the config file location configurable, then use `./configure` magic for that. 
P11SAK_DEFAULT_CONF_FILE  is already dependent on the configured OCK_CONFDIR location, so it is already configurable.... I don't see a need for any further configuration possibility.

If you really want to be able to specify the complete config file name and location during configuration, then use something like this in p11sak.h:
```
#ifdef P11SAK_DEFINED_ATTRS_LOCATION
  #define P11SAK_DEFAULT_CONF_FILE P11SAK_DEFINED_ATTRS_LOCATION
#else
  #define P11SAK_DEFAULT_CONF_FILE OCK_CONFDIR "/p11sak_defined_attrs.conf"
#endif
```
One can then use something like this to set the complete config file name and path:
`./configure  'CFLAGS=-DP11SAK_DEFINED_ATTRS_LOCATION=/path-to-file/config-file.conf'`
But I still don't see why this would be necessary at all.

Reply:
Should P11SAK_DEFINED_ATTRS_LOCATION may be an environment variable instead? If so, then you need to get its value using `getenv()`, it can't be a #define.

Reply:
Second variable removed. File location handled like opencryptoki.conf and ep11tok.conf

Reply:
In the spec you state that there would be an environment variable named `P11SAK_DEFAULT_CONF_FILE`. But I don't see that you do a `getenv()` for that variable anywhere.

Reply:
getenv() added